openedx_webhooks/info.py

```diff
@@ -35,7 +42,50 @@ def _read_repotools_file(filename):
     return resp.text
 
 def get_people_file():
-    return _read_repotools_yaml_file("people.yaml")
+    """
+    Returns data formatted as a dictionary of people containing this information:
+    {
+        github_username: {
+            name: "",
+            agreement: "",
+            institution: ""
+            jira: "",
+            other_emails: [...],
+            committer: {...},
+            comments: [...],
+            before: {...},
+            ...
+        }
+    }
+    """
+    people_data_csv = _read_repotools_csv_file("salesforce-export.csv")
+    people = dict()
+
+    for row in people_data_csv:
+        first_name = row['First Name']
+        last_name = row['Last Name']
+        acct_name = row['Account Name']
+        github_username = row['GitHub Username']
+
+        people[github_username] = {
+            "name": f"{first_name} {last_name}"
+        }
+
+        if acct_name == "Individual Contributors":
+            people[github_username]["agreement"] = 'individual'
+        elif not acct_name:
+            people[github_username]["agreement"] = "none"
+        else:
+            people[github_username]["agreement"] = 'institution'
+            people[github_username]["institution"] = acct_name
+
+    people_data_yaml = _read_repotools_yaml_file("people.yaml")
+    for p in people:
+        # Prioritzes the csv by first updating the yaml's values with csv values
+        # And then updates any missing dict fields using yaml's fields
+        people_data_yaml.get(p).update(people[p])
```

Comment:
This .get will produce None if the key is missing, which will then fail the `.update`.  You can't assume the keys in the csv all exist in the yaml, so this might throw an exception?  There's an idea for a test.

Reply:
BTW, if you do know that the key will be in the dict, then you don't need .get(), and can just use indexing, like in the next line.

Reply:
In the case that a user is in the csv, but not in the yaml, we don't want to throw an exception. We would prefer to just have the details we have from the csv. It seems like putting the dictionary updates within an if-statement checking that a user is in the yaml would bypass that.

Reply:
@nedbat But how would we test this? Is it enough to have a user in the test csv but not in the test yaml, and make sure that there aren't any yaml specific keys in people dictionary for that user?

Reply:
Yes, it's enough to add a person into the csv or yaml in the way that you want.  That's what the test data is for: it's only for our tests, so we can make it as weird and arbitrary as we need to.

Reply:
Fix and test have been added